### PR TITLE
Chapter9

### PR DESCRIPTION
**Pull request by noobsi**
chapter5

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -8,8 +9,22 @@ class User < ApplicationRecord
     uniqueness: {case_sensitive: false}
   validates :password, presence: true,
     length: {minimum: Settings.user_setting.password_minimum_length}
-  has_secure_password
   before_save :downcase_email
+  has_secure_password
+
+  def remember
+    @remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
```
Comment (thaotonto) [inline app/models/user.rb]: update_attributes

File: app/models/user.rb [hunk 2]
```diff
@@ -8,8 +9,22 @@ class User < ApplicationRecord
...
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
```
Comment (thaotonto) [inline app/models/user.rb]: same

File: app/models/user.rb [hunk 3]
```diff
@@ -20,7 +35,12 @@ def digest string
              end
```
Comment (thaotonto) [inline app/models/user.rb:35]: indent

File: app/views/static_pages/home.html.erb [hunk 4]
```diff
@@ -5,3 +5,4 @@
 </div>
 <%= link_to image_tag("rails.png", alt: "Rails logo"),
             "http://rubyonrails.org/" %>
```
Comment (thaotonto) [inline app/views/static_pages/home.html.erb]: convention

File: app/models/user.rb (hunk 3 above)
Comment (noobsi) [reply app/models/user.rb:35]: cai' nay` nhin` vao file lai. ko bi. indent

**Pull request merged**